compose/parallel.py

```diff
@@ -68,40 +74,72 @@ def _no_deps(x):
     return []
 
 
-def setup_queue(objects, func, get_deps, get_name):
+def setup_queue(objects, func, get_deps):
     if get_deps is None:
         get_deps = _no_deps
 
     results = Queue()
+    output = Queue()
+
+    t = Thread(target=queue_consumer, args=(objects, func, get_deps, results, output))
+    t.daemon = True
+    t.start()
+
+    return output
```

Comment:
Do we actually need the consumer to run in a separate thread?

It seems like it could be part of the main thread, which would let us remove the second queue.  The main thread would consume results (as it does now), and call `queue_consumer()` to launch any new threads that can be started based on the event.

The end of `queue_consumer()` looks very similar to what the main thread is doing now.

Would that work?